virttest/env_process.py

```diff
@@ -483,6 +489,11 @@ def postprocess_image(test, params, image_name, vm_process_status=None):
                 image.remove()
 
 
+def postprocess_vms(test, params, env):
+    for vm in env.get_all_vms():
```

Comment:
Is there a reason to use the environment here and the object parameters during preprocessing? Why not simply use the object parameters to improve readability from using the same approach?

Reply:
The `pre_process` uses `params` and defines all vms accordingly. Postprocess worked the same way before postprocessing all VMs defined in params, but people might create/modify the VMs dynamically throughout the execution. And many tests does that via `env_process.preprocess_vm`. They should update the params, but sometimes they forget so @xutian suggests using `env` as all VMs are usually defined there. The ones created during preprocess as well as the dynamic ones created by users in tests.

Currently users are trying to remove the dynamically created VMs in various try/finally blocks, but this should be way safer. Just grep the `io-github-autotest-qemu` for `preprocess_vm` to see how many tests are doing that.

Reply:
Btw with this patch we could significantly cleanup the error-handing in those tests and theoretically in migration as well as the clone VM is also registered in env and would be cleaned automatically.

Reply:
I see. I assume using the `env` during preprocessing as an alternative clashes with the perception of temporary envoronment that many developers have. In that case the change is fine, perhaps with a little comment on this clarified difference so that future looks at this code don't wonder too.

Reply:
As https://github.com/avocado-framework/avocado-vt/pull/2105 was merged this needs to be conditional `keep_unrequested_vms = params.get_boolean("keep_env_vms", "no")`